packages/runtime/test/integration/group-inner-transaction.ts

```diff
@@ -74,4 +76,89 @@ describe("Group inner transaction", function () {
 		// spend 5000 micro Algo
 		assert.equal(contractBalance, contractAcc.balance() + 5000n);
 	});
+
+	describe("Number inner transaction limitation", function () {
+		let secondAppInfo: AppInfo;
+		this.beforeEach(() => {
+			secondAppInfo = runtime.deployApp(
+				alice.account,
+				{
+					appName: "secondApp",
+					metaType: types.MetaType.FILE,
+					approvalProgramFilename: "limit-number-txn.py",
+					clearProgramFilename: "clear.teal",
+					globalBytes: 1,
+					globalInts: 1,
+					localBytes: 1,
+					localInts: 1,
+				},
+				{}
+			);
+
+			const txnParams: types.AlgoTransferParam = {
+				type: types.TransactionType.TransferAlgo,
+				sign: types.SignType.SecretKey,
+				fromAccount: alice.account,
+				toAccountAddr: secondAppInfo.applicationAccount,
+				amountMicroAlgos: 1e6,
+				payFlags: {
+					totalFee: 1000,
+				},
+			};
+
+			runtime.executeTx([txnParams]);
+			syncAccounts();
+		});
+
+		it("Should fail when issue more than 256 inner txn", () => {
```

Comment:
@vuvoth Will this fail, here are passing only 4 txn?

Reply:
No this is more than 256 inner transaction. You can see one inner payment is one inner and we issue 16 * 4 for one single transaction => total transaction is (4 + 16 * 4 * 4) > 256 ( not include big one)